src/loaders/utils.js

```diff
@@ -102,3 +104,75 @@ export function isBioformatsNoPadHeightVersion(software) {
   }
   return false;
 }
+
+/**
+ * Returns actual image stats for static imagery and an estimate via a downsampled version of image pyramids.
+ * This is helpful for generating histograms of your channel data, or scaling your sliders down to a reasonable range.
+ * @param {Object} args
+ * @param {Object} args.loader A valid loader object.
+ * @param {Array} args.loaderSelection Array of valid dimension selections
+ * @returns {Object} { means, dataRanges, standardDeviations, data, firstQuartiles, thirdQuartiles }, each arrays with entries for each channel based on your selection.
+ */
+export async function getChannelStats({ loader, loaderSelection }) {
+  const z = loader.isPyramid ? loader.numLevels - 1 : 0;
+  const rasters = await loader.getRaster({ z, loaderSelection });
+  const { data } = rasters;
+  const channelStats = {
+    means: [],
+    medians: [],
+    dataRanges: [],
+    standardDeviations: [],
+    data,
+    firstQuartiles: [],
+    thirdQuartiles: []
+  };
+  data.forEach(arr => {
+    let len = arr.length;
+    let min = Infinity;
+    let max = -Infinity;
+    let total = 0;
+    // Range (min/max).
+    // eslint-disable-next-line no-plusplus
+    while (len--) {
+      if (arr[len] < min) {
+        min = arr[len];
+      }
+      if (arr[len] > max) {
+        max = arr[len];
+      }
+      total += arr[len];
+    }
+    channelStats.dataRanges.push([min, max]);
+
+    // Mean.
+    const mean = total / arr.length;
+    channelStats.means.push(mean);
+
+    // Standard Deviation.
+    len = arr.length;
+    let sumSquared = 0;
+    // eslint-disable-next-line no-plusplus
+    while (len--) {
+      sumSquared += (arr[len] - mean) ** 2;
+    }
+    const standardDeviation = (sumSquared / arr.length) ** 0.5;
+    channelStats.standardDeviations.push(standardDeviation);
+
+    // Median, and quartiles via quickselect: https://en.wikipedia.org/wiki/Quickselect.
+    // Odd number lengths should round down the index.
+    const mid = Math.floor(arr.length / 2);
+    const firstQuartileLocation = Math.floor(arr.length / 4);
+    const thirdQuartileLocation = 3 * Math.floor(arr.length / 4);
```

Comment:
Should use `//` divide?

Reply:
I think this just creates a comment, unless there is some other way to use it.